apply-roles

```diff
@@ -279,28 +357,54 @@ main() {
   readonly UPDATE_ROLE
 
   if [[ "${UPDATE_ROLE}" == 'false' ]]; then
-    read -rp "Would you like to apply \"${CUSTOM_ROLE_NAME}\" to a all subscriptions [Y|N]? " answer
+    read -rp "Would you like to apply \"${CUSTOM_ROLE_NAME}\" to all subscriptions [Y|N]? " answer
     case "${answer}" in
     [yY] | [yY][eE][sS])
+      APPLY_ROLE='true'
       echo "\"${CUSTOM_ROLE_NAME}\" will be applied to all subscriptions."
       ;;
     [nN] | [nN][oO])
-      while true; do
-        read -rp "Please type in the subscription id: " subscription_id
-        subscription_name=$(az account list --query "[? id == '$subscription_id'].name" --output tsv)
-        if [[ -n "${subscription_name}" ]]; then
-          echo "Subscription \"${subscription_name}\" found."
-          break
-        fi
-        echo "Subscription wasn't found. Please type in the correct subscription id."
-      done
+      read -rp "Would you like to apply \"${CUSTOM_ROLE_NAME}\" to specific subscription [Y|N]? " answer
+      case "${answer}" in
+      [yY] | [yY][eE][sS])
+        APPLY_ROLE='true'
```

Comment:
🤔 I'm assuming that you added this flag in so that a user could skip adding the subscription level customer role but still apply the tenant level role below.

If a customer only wanted to apply the tenant level role then they would have to know to select no for both of these scenarios. Not the end of the world but not very intuitive... this might not be a big deal but is worth thinking about the user experience of a customer that just wants to update that 1 permission?

Reply:
Yes, the `APPLY_ROLE` was introduced to let the customer "opt out" of updating subscription level customer role.

Customer would be prompted to make a decision for the tenant scope custom role regardless of their prior choices, it is a choice at the root level not nested under anything and has its own variable to keep track of the decision, `UPSERT_TENANT_ROLE`. 

But you are correct that there is an oversight - I realize that the `UPDATE_ROLE` decision however would render the `UPSERT_TENANT_ROLE` decision useless. I have moved the question after the `UPDATE_ROLE` choice.

Reply:
I thought about this after perhaps we could have some higher level questions first?

```
Would you like to apply the custom role to the tenant? Y/N
Would you like to apply the custom role to subscriptions? Y/N
```

Then can still have the questions about applying to all subscriptions or just 1. Using this approach customers that already have the role setup for subscriptions don't need to go through that part of the script at all?

Reply:
Got it, adjusted accordingly. Is this what you had in mind?

https://github.com/cloudconformity/azure-onboarding-scripts/pull/29/commits/cbb10569e75938747cbbaafb2aaddae1c414a406#diff-7a1895416f0c412b1b211e7300d7c395b09f62d30d81097b9cebbeeabaa57752R361-R408

Reply:
Not exactly what I was suggesting, but I like your implementation better 🙂 🚀